### ISSUE TITLE
Add ASTC compression as an extension

### ISSUE THREAD

**Issue by britzl** [defold/defold#5938]
**Is your feature request related to a problem? Please describe (REQUIRED):**
WebP compression was removed when the texture compression pipeline was migrated to Basis Universal. WebP (especially WebP Lossy) still has some merit in an HTML5 game context where compressed texture size is very important to keep game size as small as possible.

**Describe the solution you'd like (REQUIRED):**
WebP compression should be brought back as an extension

**Describe alternatives you've considered (REQUIRED):**
Could WebP compressed images be loaded completely "on the side" somehow? Can a fully configurable Basis Universal compression pipeline reach similar results as WebP Lossy?

**Additional context (OPTIONAL):**
Discussed on the forum:

* https://forum.defold.com/t/defold-1-2-185-beta/68840/6?u=britzl
* https://forum.defold.com/t/texture-compression-update-alpha-testing/67470/24?u=britzl

Sub tasks:
* [x] #9820
* [x] #9823
* [x] #9821

**Comment by AGulev** [comment]
yeah, in Ru community many people use Defold for web games and osme of them decided to stay with 1.2.179 because it has WebP.

**Comment by taracheshnikov** [comment]
I think it would be great to add WebP support

**Comment by britzl** [comment]
> I think it would be great to add WebP support

Remember to give it a 👍  on the first post of this issue!

**Comment by dotnetmouse** [comment]
As of 2023, WebP compression is still not supported?

**Comment by britzl** [comment]
> As of 2023, WebP compression is still not supported?

Correct. But we are now pretty sure of the path to an implementation. I think this will be one of the upcoming tasks for us to work on early in 2024.

**Comment by AGulev** [comment]
Using png maybe a good option as well (since we already have `image` lib in the engine)

**Comment by JCash** [comment]
Note that I created a separate ticket for being able to support WebP: https://github.com/defold/defold/issues/9955

**Comment by Jhonnyg** [comment]
This is done, with the caveat that ASTC support isn't an extension - it's part of the Defold core instead.
